service/common/src/main/java/org/apache/polaris/service/auth/JWTSymmetricKeyFactory.java

```diff
@@ -19,56 +19,58 @@
 package org.apache.polaris.service.auth;
 
 import io.smallrye.common.annotation.Identifier;
+import jakarta.annotation.Nullable;
 import jakarta.inject.Inject;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
+import java.util.Objects;
 import java.util.function.Supplier;
 import org.apache.polaris.core.context.RealmContext;
 import org.apache.polaris.core.persistence.MetaStoreManagerFactory;
 
 @Identifier("symmetric-key")
 public class JWTSymmetricKeyFactory implements TokenBrokerFactory {
-  @Inject private MetaStoreManagerFactory metaStoreManagerFactory;
-  private int maxTokenGenerationInSeconds = 3600;
-  private String file;
-  private String secret;
+
+  private final MetaStoreManagerFactory metaStoreManagerFactory;
+  private final Config config;
+
+  @Inject
+  public JWTSymmetricKeyFactory(MetaStoreManagerFactory metaStoreManagerFactory, Config config) {
+    this.metaStoreManagerFactory = metaStoreManagerFactory;
+    this.config = config;
+  }
 
   @Override
   public TokenBroker apply(RealmContext realmContext) {
-    if (file == null && secret == null) {
+    String secret = config.secret();
```

Comment:
I guess this check can be moved to the constructor.

Reply:
I'm for checking in the constructor, FWIW

Reply:
ok, will update

Reply:
refactored (and rebased)